src/core/operations/XMLToJSON.mjs

```diff
@@ -0,0 +1,49 @@
+/**
+ * @author jpledref [jp.ledref@gmail.com]
+ * @copyright Crown Copyright 2023
+ * @license Apache-2.0
+ */
+
+import xml2js from "xml2js";
+import Operation from "../Operation.mjs";
+import OperationError from "../errors/OperationError.mjs";
+
+/**
+ * XML to JSON operation
+ */
+class XMLToJSON extends Operation {
+
+    /**
+     * XMLToJSON constructor
+     */
+    constructor() {
+        super();
+
+        this.name = "XML to JSON";
+        this.module = "Default";
+        this.description = "Converts XML data to JSON format.";
+        this.infoURL = "https://wikipedia.org/wiki/XML";
+        this.inputType = "string";
+        this.outputType = "JSON";
+        this.args = [
+        ];
+    }
+
+    /**
+     * @param {string} input
+     * @param {Object[]} args
+     * @returns {JSON}
+     */
+    run(input, args) {
+        let result;
+        try {
+            xml2js.parseString(input, {}, (e, r) => result = JSON.stringify(r));
```

Comment:
You're not using the error result in the callback. I assume xml2js uses that rather than throwing an error, and it's going to include useful parser errors.